### DISCUSSION TITLE
Embedding browser resources in Scalar.AspNetCore NuGet package

### DISCUSSION THREAD

**Discussion by mkstephenson** [scalar/scalar#3357] [Ideas]
Hi,

I am currently looking at adding the .NET version of the package to a project, however the dependency on external resources via the CDN makes it impossible to use this in the project.

Would it be possible to deliver a version of the NuGet package with the JS, fonts and CSS embedded (in the same way the Swagger UI package does that)? That way we would be able to use the wonderful UI on systems where connections to external CDNs are not available.

**Comment by xC0dex** [comment]
Hey @mkstephenson,
this is a great idea! Embedding the all static files directly in the `Scalar.AspNetCore` package would be possible to implement, but there are some open questions:

- Should we always update the .NET package whenever a new API reference is released? Otherwise, the package may not use the latest version.
- Should we make this opt-in? So, by default the latest CDN will be used but with an option to switch to the offline version?

Any thoughts @marclave @hanspagel?

**Comment by xC0dex** [comment]
Hey @mkstephenson, we just released version `1.2.10` that uses local assets by default. 🎉

**Reply by mkstephenson** [reply]
Great, thanks!
